### PR TITLE
[16.0][IMP] stock_picking_auto_create_lot

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
This PR moves `_get_lot_sequence` method from `stock.picking` to` stock.move.line`. This change allows us to extend the method to retrieve the specific sequence based on the product of each move line.
Related: https://github.com/OCA/stock-logistics-workflow/pull/1619

@qrtl QT4624

### CODE REVIEW

File: stock_picking_auto_create_lot/models/stock_move_line.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Copyright 2024 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    @api.model
```
Comment (yostashiro) [inline stock_picking_auto_create_lot/models/stock_move_line.py]: I think it's more appropriate not to have the decorator. We expect to work on a given record when it's extended.
```suggestion
```

**Review by yostashiro — APPROVED** [review]
Code review.

**Review by sergio-teruel — APPROVED** [review]

**Review by rousseldenis — APPROVED** [review]

**Comment by AungKoKoLin1997** [comment]
@rousseldenis  Can we get this PR merged?

**Pull request merged**